### PR TITLE
[Frontend] Support embeddings in the run_batch API

### PR DESCRIPTION
**Pull request by pooyadavoodi**
Currently run_batch which is the offline batching API only supports chat completion. This PR adds embedding supports to run_batch. The PR also adds support for empty lines (or lines with only whitespace) in the input file of run_batch.

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]
Looks great, do you mind adding a simple example to `examples/offline_inference_openai.md`?

**Comment by pooyadavoodi** [comment]
> Looks great, do you mind adding a simple example to `examples/offline_inference_openai.md`?

Thanks @wuisawesome 
I updated the md file. 
I wrote that `vllm >= 0.5.5` is required as that's the next version that will be released. Is that okay?

**Comment by wuisawesome** [comment]
> I wrote that vllm >= 0.5.5 is required as that's the next version that will be released. Is that okay?

Yep!

File: examples/offline_inference_openai.md [hunk 1]
```diff
@@ -170,3 +170,36 @@ Your results are now on S3. You can view them in your terminal by running
...
...65625,0.0234375,0.005764007568359375,-0.0075836181640625,0.0229949951171875,-0.031951904296875,-0.00933837890625,0.01232147216796875,-0.000919342041015625,0.01192474365234375,0.0264129638671875,0.03338623046875,0.01983642578125,-0.025146484375,0.0004684925079345703,-0.002651214599609375,0.013824462890625,-0.025115966796875,0.005817413330078125,0.003631591796875,-0.005855560302734375,0.00024199485778808594,-0.00820159912109375,-0.007274627685546875,-0.00287628173828125,0.0021114349365234375,0.0290679931640625,-0.01361083984375,-0.0135955810546875,0.0192108154296875,0.0229644775390625,-0.0029163360595703125,0.0157928466796875,-0.0019817352294921875,0.0089874267578125,-0.01247406005859375,0.0158843994140625,0.0210113525390625,-0.00823211669921875,-0.0034427642822265625,0.005268096923828125,-0.0007991790771484375,0.00762939453125,-0.002971649169921875,0.0240020751953125,0.0027923583984375,0.006160736083984375,0.012115478515625,-0.01149749755859375,-0.005985260009765625,-0.023773193359375,0.0073699951171875,0.00977325439453125,0.0110626220703125,0.0223236083984375,-0.00991058349609375,0.025360107421875]}],"usage":{"prompt_tokens":8,"total_tokens":8,"completion_tokens":0}}},"error":null}
```
Comment (simon-mo) [inline examples/offline_inference_openai.md:203]:
```suggestion
{"id":"vllm-db0f71f7dec244e6bce530e0b4ef908b","custom_id":"request-1","response":{"status_code":200,"request_id":"vllm-batch-3580bf4d4ae54d52b67eee266a6eab20","body":{"id":"embd-33ac2efa7996430184461f2e38529746","object":"list","created":444647,"model":"intfloat/e5-mistral-7b-instruct","data":[{"index":0,"object":"embedding","embedding":[0.016204833984375,0.0092010498046875,0.0018358230590820312,-0.0028228759765625,0.001422882080078125,-0.0031147003173828125,...]}],"usage":{"prompt_tokens":8,"total_tokens":8,"completion_tokens":0}}},"error":null}
```

File: examples/offline_inference_openai.md [hunk 2]
```diff
@@ -170,3 +170,36 @@ Your results are now on S3. You can view them in your terminal by running
...
....0091094970703125,0.0309295654296875,-0.0072021484375,-0.002048492431640625,0.037078857421875,-0.0113372802734375,-0.0021190643310546875,0.0096893310546875,-0.00041294097900390625,0.017791748046875,0.0288848876953125,0.0232696533203125,0.0160064697265625,-0.01666259765625,-0.0047760009765625,-0.002349853515625,0.00743865966796875,-0.02313232421875,0.01284027099609375,0.006214141845703125,0.005077362060546875,-0.00806427001953125,-0.005359649658203125,0.005908966064453125,-0.0014047622680664062,-0.004894256591796875,0.02203369140625,-0.0042877197265625,-0.004669189453125,0.0157623291015625,0.0143890380859375,-0.0006694793701171875,0.005870819091796875,0.009796142578125,0.00836944580078125,-0.010711669921875,0.00628662109375,0.02117919921875,-0.0172271728515625,-0.009033203125,0.00334930419921875,0.010162353515625,0.00646209716796875,0.00045990943908691406,0.02008056640625,0.0010156631469726562,0.0074615478515625,0.00897216796875,-0.0116729736328125,-0.00714874267578125,-0.036041259765625,0.011077880859375,-0.0282135009765625,0.00817108154296875,0.027679443359375,-0.022705078125,0.0249176025390625]}],"usage":{"prompt_tokens":10,"total_tokens":10,"completion_tokens":0}}},"error":null}
```
Comment (simon-mo) [inline examples/offline_inference_openai.md:204]:
```suggestion
...```

**Review by simon-mo — APPROVED** [review]

**Pull request merged**